### ISSUE TITLE
Some questions

### ISSUE THREAD

**Issue by yyEudora** [MMorafah/PACFL#2]
Hello, Dr. MMorafah, I ran your code recently and found that the final result is shown below, which cannot reach the classification accuracy of 89.30 mentioned in your paper. So I want to ask which set of parameters can be set to achieve the classification accuracy you mentioned. In addition, I also want to ask, does 20% label skew mean that each client only has two categories of data samples? I see that the default parameter you gave is 'noniid-#label2', does this correspond to the 20% label skew mentioned in the paper?

`Train Loss: 0.000535977038280345, Test_loss: 0.8078835224452473
Train Acc: 100.0, Test Acc: 84.48151397705078
Best Clients AVG Acc: 85.41549682617188
Cluster 0, Best Glob Acc 41.250
Cluster 1, Best Glob Acc 39.110
Average Best Glob Acc 40.180`

**Comment by MMorafah** [comment]
Hello,

'noniid-#label2’ is label skew 20%. You may need to work with the hyper-parameters to get closer results. I don’t remember the exact set of hyper-parameters. Also, your results seem to be for a single run, and there is some randomness in the results.

Regards,
Mahdi

**Comment by yyEudora** [comment]
> Hello, 'noniid-#label2’ is label skew 20%. You may need to work with the hyper-parameters to get closer results. I don’t remember the exact set of hyper-parameters. Also, your results seem to be for a single run, and there is some randomness in the results. Regards, Mahdi
> […](#)
> On Jun 17, 2023 at 4:51 AM -0700, yyEudora ***@***.***>, wrote: Hello, Dr. MMorafah, I ran your code recently and found that the final result is shown below, which cannot reach the classification accuracy of 89.30 mentioned in your paper. So I want to ask which set of parameters can be set to achieve the classification accuracy you mentioned. In addition, I also want to ask, does 20% label skew mean that each client only has two categories of data samples? I see that the default parameter you gave is 'noniid-#label2', does this correspond to the 20% label skew mentioned in the paper? Train Loss: 0.000535977038280345, Test_loss: 0.8078835224452473 Train Acc: 100.0, Test Acc: 84.48151397705078 Best Clients AVG Acc: 85.41549682617188 Cluster 0, Best Glob Acc 41.250 Cluster 1, Best Glob Acc 39.110 Average Best Glob Acc 40.180 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

Thank you for your warm reply. I would like to know whether the results in the paper are the results of a certain experiment or the average of the results of multiple experiments?

**Comment by MMorafah** [comment]
Hello,

The results are averaged over multiple runs (3 different runs).

Regards,
Mahdi

**Comment by yyEudora** [comment]
> Hello, The results are averaged over multiple runs (3 different runs). Regards, Mahdi
> […](#)
> On Jun 19, 2023 at 8:23 PM -0700, yyEudora ***@***.***>, wrote: > Hello, 'noniid-#label2’ is label skew 20%. You may need to work with the hyper-parameters to get closer results. I don’t remember the exact set of hyper-parameters. Also, your results seem to be for a single run, and there is some randomness in the results. Regards, Mahdi > … > On Jun 17, 2023 at 4:51 AM -0700, yyEudora @.>, wrote: Hello, Dr. MMorafah, I ran your code recently and found that the final result is shown below, which cannot reach the classification accuracy of 89.30 mentioned in your paper. So I want to ask which set of parameters can be set to achieve the classification accuracy you mentioned. In addition, I also want to ask, does 20% label skew mean that each client only has two categories of data samples? I see that the default parameter you gave is 'noniid-#label2', does this correspond to the 20% label skew mentioned in the paper? Train Loss: 0.000535977038280345, Test_loss: 0.8078835224452473 Train Acc: 100.0, Test Acc: 84.48151397705078 Best Clients AVG Acc: 85.41549682617188 Cluster 0, Best Glob Acc 41.250 Cluster 1, Best Glob Acc 39.110 Average Best Glob Acc 40.180 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you are subscribed to this thread.Message ID: @.> Thank you for your warm reply. I would like to know whether the results in the paper are the results of a certain experiment or the average of the results of multiple experiments? — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***>

I see, thank you very much!

**Comment by yyEudora** [comment]
Hello, Dr. MMorafah, 
Through many experiments and comparisons, I recently found that for the cifar10 dataset, when the client is divided into a cluster, the average personalization performance of the 100 client models is better than that when the client divided into multiple clusters. Whereas what you describe in your article is,  when there is only one cluster, the personalization performance of the global model is focused on. What I want to ask is why not pay attention to the average personalization performance of 100 client models when clients are divided into one cluster?

**Comment by MMorafah** [comment]
Hello,

When there is only one cluster, it means that all clients are being optimized together (no personalization). You might need to take a look at the clusters formed. Every time you run the code, a new set of clients are being formed with a new partitioning and the clustering threshold might be different for that partitioning. I suggest, fixating the clients for your experiments.

Regards,
Mahdi

**Comment by yyEudora** [comment]
Thank you, I get it.

**Comment by yangtao101** [comment]
Hello, Dr. MMorafah
 How to fix the number of clusters and fixating the clients for my experiments?

**Comment by MMorafah** [comment]
Hello,

Once you generate the data partitions for the clients, you can simply save the data partitions and load them for your experiment. You may also try fixating the random seed which will result into the same partition generation for each client.

Regards,
Mahdi
